File: test/Models/energybalance_pkg/src/cpp/EnergybalanceState.cpp
#include "EnergybalanceState.h"
EnergybalanceState::EnergybalanceState() { }

float EnergybalanceState::getdiffusionLimitedEvaporation() {return this-> diffusionLimitedEvaporation; }
float EnergybalanceState::getconductance() {return this-> conductance; }
float EnergybalanceState::getminCanopyTemperature() {return this-> minCanopyTemperature; }
float EnergybalanceState::getmaxCanopyTemperature() {return this-> maxCanopyTemperature; }

void EnergybalanceState::setdiffusionLimitedEvaporation(float _diffusionLimitedEvaporation) { this->diffusionLimitedEvaporation = _diffusionLimitedEvaporation; }
void EnergybalanceState::setconductance(float _conductance) { this->conductance = _conductance; }
void EnergybalanceState::setminCanopyTemperature(float _minCanopyTemperature) { this->minCanopyTemperature = _minCanopyTemperature; }
void EnergybalanceState::setmaxCanopyTemperature(float _maxCanopyTemperature) { this->maxCanopyTemperature = _maxCanopyTemperature; }

File: test/Models/energybalance_pkg/src/java/EnergybalanceState.java
import  java.io.*;
import  java.util.*;
public class EnergybalanceState
{
    private double diffusionLimitedEvaporation;
    private double conductance;
    private double minCanopyTemperature;
    private double maxCanopyTemperature;
    
    public EnergybalanceState()
    {
           
    }
    
    public EnergybalanceState(EnergybalanceState toCopy, boolean copyAll) // copy constructor 
    {
        if (copyAll)
        {
            this.diffusionLimitedEvaporation = toCopy.diffusionLimitedEvaporation;
            this.conductance = toCopy.conductance;
            this.minCanopyTemperature = toCopy.minCanopyTemperature;
            this.maxCanopyTemperature = toCopy.maxCanopyTemperature;
        }
    }
    public double getdiffusionLimitedEvaporation()
    {
        return diffusionLimitedEvaporation;
    }

    public void setdiffusionLimitedEvaporation(double _diffusionLimitedEvaporation)
    {
        this.diffusionLimitedEvaporation= _diffusionLimitedEvaporation;
    } 
    
    public double getconductance()
    {
        return conductance;
    }

    public void setconductance(double _conductance)
    {
        this.conductance= _conductance;
    } 
    
    public double getminCanopyTemperature()
    {
        return minCanopyTemperature;
    }

    public void setminCanopyTemperature(double _minCanopyTemperature)
    {
        this.minCanopyTemperature= _minCanopyTemperature;
    } 
    
    public double getmaxCanopyTemperature()
    {
        return maxCanopyTemperature;
    }

    public void setmaxCanopyTemperature(double _maxCanopyTemperature)
    {
        this.maxCanopyTemperature= _maxCanopyTemperature;
    } 
    
}